internal/controller/pdbwatcher_controller.go

```diff
@@ -102,14 +122,52 @@ func (r *PDBWatcherReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 	deployment := &appsv1.Deployment{}
 	err = r.Get(ctx, types.NamespacedName{Name: deploymentName, Namespace: pdbWatcher.Namespace}, deployment)
 	if err != nil {
-		return ctrl.Result{}, err
+		return ctrl.Result{}, err // Error fetching Deployment
 	}
 
+	// Check if the resource version has changed
```

Comment:
better comment on why we're doing this talk about if someone else has changed the deployment we go "hands off"